utils/dataloaders.py

```diff
@@ -0,0 +1,92 @@
+import random
+
+import torch
+import numpy as np
+from omegaconf import DictConfig
+from datasets import load_dataset
+from torch.utils.data import Dataset, DataLoader
+
+
+class CustomECGDataset(Dataset):
+    def __init__(self, hf_dataset):
+        self.hf_dataset = hf_dataset
+
+    def __len__(self):
+        return len(self.hf_dataset) * 4  # Each original sample will produce 4 new samples
+
+    def __getitem__(self, index):
+        original_index = index // 4  # Determine the index in the original dataset
+        slice_index = index % 4  # Determine which slice (0, 1, 2, or 3)
+
+        sample = self.hf_dataset[original_index]
+        start = slice_index * 250
+        end = start + 250
+
+        channel1 = torch.tensor(sample["signal"][0][start:end])
+        channel2 = torch.tensor(sample["signal"][1][start:end])
+
+        return channel1, channel2
+
+
+class CombinedECGDataset(Dataset):
+    def __init__(self, *datasets):
```

Comment:
```suggestion
    def __init__(self, dataset: HFDataset):
```

It's annoying that we have to use a classed called `Dataset` from two different libraries, but this is the way.